src/api/firebase.js

```diff
@@ -80,11 +82,38 @@ export async function addItem(listId, { itemName, daysUntilNextPurchase }) {
 
 export async function updateItem(listId, itemId) {
 	const docRef = doc(db, listId, itemId);
+	// We need to use some data from the document for our future purchase estimates.
+	const docSnap = await getDoc(docRef);
+	const today = new Date();
 
-	await updateDoc(docRef, {
-		dateLastPurchased: new Date(),
-		totalPurchases: increment(1),
-	});
+	if (docSnap.exists()) {
+		const data = docSnap.data();
+		// If the user has never purchased this item before, we'll use the date
+		// it was created as the date of the last purchase.
+		const daysBetweenPurchases = getDaysBetweenDates(
+			data.dateLastPurchased?.toDate() ?? data.dateCreated.toDate(),
```

Comment:
Nice use of both the optional chaining operator and nullish coalescing operator. I had to look into what they are/do, but definitely does make the code cleaner.